dart/lib/src/sentry_client.dart

```diff
@@ -131,18 +132,42 @@ class SentryClient {
       return event;
     }
 
-    if (event.exceptions?.isNotEmpty ?? false) return event;
+    if (event.exceptions?.isNotEmpty ?? false) {
+      return event;
+    }
+
+    final isolateName = getIsolateName();
+    // isolates have no id, so we use the hashCode as id
+    final isolateId = isolateName?.hashCode;
```

Comment:
Isolates have no id, so I'm using the hashcode of the name, in order to get a unique id in order to link thread and exception.

I believe the same approach has to be taken for the Java/Kotlin exception reporting from the PlatformException as the thread id is not reported in a stringified stracktrace.